src/OpenTelemetry.Collector.AspNet/Implementation/HttpInListener.cs

```diff
@@ -0,0 +1,143 @@
+﻿// <copyright file="HttpInListener.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Web;
+using System.Web.Routing;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Collector.AspNet.Implementation
+{
+    internal class HttpInListener : ListenerHandler
+    {
+        private readonly PropertyFetcher routeFetcher = new PropertyFetcher("Route");
+        private readonly PropertyFetcher routeTemplateFetcher = new PropertyFetcher("RouteTemplate");
+        private readonly AspNetCollectorOptions options;
+
+        public HttpInListener(string name, Tracer tracer, AspNetCollectorOptions options)
+            : base(name, tracer)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnStartActivity";
```

Comment:
entire method should be under a big try{}catch. I commented on one place that may throw, but there may be more. And better to have an error message logged.

Reply:
It looks like `DiagnosticSourceListener.OnNext` (parent) wraps the call to OnStartActivity in a try/catch w/ a log, you still want me to add a specific one here?